zebra: Get link from the correct netns for vxlan

leonshaw opened the pull request:
If VRF backend is netns, bridges and SVIs are in the VRF's netns.
Get link information from there, rather than NS_DEFAULT.

leonshaw committed:
zebra: Get link from the correct netns for vxlan

If VRF backend is netns, bridges and SVIs are in the VRF's netns.
Get link information from there, rather than NS_DEFAULT.

Signed-off-by: Xiao Liang <shaw.leon@gmail.com>

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/3a197daf037246e29f9818d19414be2d/raw/08ce2bd16556525e398625a668c26e0c1a25362b/cr_8895_1624328326.diff | git apply
```

</p>

<p>

```diff
diff --git a/zebra/zebra_vxlan.c b/zebra/zebra_vxlan.c
index 8a7b12c0b..873e4e462 100644
--- a/zebra/zebra_vxlan.c
+++ b/zebra/zebra_vxlan.c
@@ -2032,7 +2032,8 @@ static int zebra_vxlan_handle_vni_transition(struct zebra_vrf *zvrf, vni_t vni,
 				   vni);
 
 		/* Find VxLAN interface for this VNI. */
-		for (rn = route_top(zvrf->zns->if_table); rn; rn = route_next(rn)) {
+		for (rn = route_top(zvrf->zns->if_table); rn;
+		     rn = route_next(rn)) {
 			ifp = (struct interface *)rn->info;
 			if (!ifp)
 				continue;

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

leonshaw commented:
Rebased against master.
Removed `zl3vni_from_svi()` part as is resolved in #9416 by a walk through all namespaces. However, I think the walk could be avoided since the vxlan interface is in the same VRF of the bridge, which is already known (see original commit of this PR).

leonshaw commented:
ci:rerun

leonshaw commented:
@pjdruddy It's long since this PR was open, any comments?

@pguibert6WIND Could have look at this maybe?

leonshaw commented:
rebased

ton31337 commented:
@Mergifyio rebase

ton31337 reviewed (approved):
LGTM

ton31337 commented:
@chiragshah6 would you like to take a look at this before merging?

riw777 commented:
@greptile review

riw777 reviewed (approved):
looks good
